src/carma-streets/parse_ros_bags.py

```diff
@@ -1,51 +1,90 @@
-import rosbag
-import csv
-import pandas as pd
-import matplotlib.pyplot as plt
 import argparse
+import csv
+import itertools
+import math
 from pathlib import Path
 
+import rosbag
+
+
+def find_msg_closest_to_timestamp(bag_msgs, timestamp):
+    def pairwise(iterable):
```

Comment:
Is this not already externally available using https://docs.python.org/3.8/library/itertools.html the `more-itertools` pip package

Reply:
Either way, maybe at least include a reference to the source documentation

Reply:
I didn't see the `more-itertools` package when I was looking through the docs. I will update.